### PR TITLE
Safehouse selection and pathfinding grid

### LINKED ISSUE #16: Safehouse selection and pathfinding grid
## Objective
Implement the safehouse selection algorithm that chooses the best defensible building for the player, and build the PathFinding.js navigation grid from the generated tilemap for zombie AI pathfinding.

## Acceptance Criteria
- [ ] Candidate buildings scored by: number of entry points (fewer = easier to defend), proximity to loot-rich areas, building size, pre-existing useful objects
- [ ] Highest-scoring candidate designated as safehouse with a distinct visual indicator (colored border or highlight)
- [ ] Safehouse entry points (doors, windows) identified and stored for barricading mechanics
- [ ] PathFinding.js grid constructed from the generated tilemap — walls are non-walkable, doors are walkable, open spaces are walkable
- [ ] Grid update API exposed for dynamic changes when barricades are placed or destroyed
- [ ] A* pathfinding verified: paths can be computed from arbitrary city points to the safehouse
- [ ] Safehouse location marked on the minimap data (consumed by HUD in M5)

## Technical Notes
- PathFinding.js uses a 2D grid where each cell is walkable or blocked — map directly from the tilemap
- The grid must be updatable at runtime because barricade placement changes walkability
- Safehouse scoring weights should be tunable constants, not hardcoded magic numbers
- Zombie spawn zones should be placed at map edges and in buildings far from the safehouse
- Store spawn zone locations for the wave spawner system in M3

## Dependencies
- Object placement and loot tables

### PR DESCRIPTION
**Pull request by allan-mobley-jr**
## Summary
Closes #16: Implement safehouse selection algorithm, PathFinding.js navigation grid, and spawn zone placement for zombie AI pathfinding.

## Changes
- Shared procgen type definitions (`CityLayout`, `Building`, `EntryPoint`, `SafehouseResult`, `PathResult`, `SpawnZone`)
- Tunable constants for safehouse scoring weights, spawn zone config, and pathfinding parameters
- Shared geometry utilities (`tileDistance`, `getBuildingCenter`)
- Weighted safehouse scoring algorithm (entry points, loot proximity, building size, object density)
- PathfindingGrid class wrapping PathFinding.js with runtime walkability updates for barricading
- Edge and far-building spawn zone placement with seeded PRNG determinism
- Temperer-approved fixes: NO_PATH factory (mutation safety), shared `isWalkableTileType` (enum exhaustiveness), sealed-building filter (unreachable safehouse prevention)

## Test Coverage
- Created `geometry.test.ts` — 6 tests for shared tile-space utilities
- Added to `safehouse.test.ts` — 5 tests: sizeFitness boundary values (IDEAL_SIZE_MIN, IDEAL_SIZE_MAX, 2x max, zero area), Container category classification (excluded from loot, included in cover)
- Added to `pathfinding-grid.test.ts` — 1 test: smoothed paths produce fewer waypoints than raw A*
- Added to `spawn-zones.test.ts` — 4 tests: sealed building uses center fallback, entry point preferred over center, edge zone discard on insufficient walkable points, zero zones on all-wall edges
- Total: 748 tests passing (16 new, 30 test files)

## Acceptance Criteria
- [x] Candidate buildings scored by: entry points, loot proximity, building size, pre-existing objects
- [x] Highest-scoring candidate designated as safehouse with minimap position
- [x] Safehouse entry points identified and stored for barricading mechanics
- [x] PathFinding.js grid constructed from generated tilemap
- [x] Grid update API exposed for dynamic changes (barricade placement/destruction)
- [x] A* pathfinding verified: paths computed from map edges to safehouse through doors
- [x] Safehouse location marked on minimap data via `toMatrix()` export

## Quality Checks
- [x] Lint passes (0 errors, 2 pre-existing warnings)
- [x] Type check passes
- [x] Tests pass (748 total)
- [x] Build succeeds

### CODE REVIEW

File: src/game/procgen/spawn-zones.ts [hunk 1]
```diff
@@ -49,11 +49,7 @@ export function findWalkableSpawnPoints(
...
 
       const tile = cityLayout.tiles[y]?.[x];
-      if (
-        tile !== undefined &&
-        tile !== TileType.Wall &&
-        tile !== TileType.Empty // spawn on roads/sidewalks/floors, not void
-      ) {
+      if (tile !== undefined && isWalkableTileType(tile)) {
         points.push({ x, y });
       }
```
[the remark this answers is not in the document]
Comment (allan-mobley-jr) [reply src/game/procgen/spawn-zones.ts]: This is correct by design — windows are intentionally walkable tile types. They are entry points (like doors) that zombies can path through and attack. There is no separate "runtime tile collision table"; the `isWalkableTileType` exhaustive switch IS the walkability source of truth, and it deliberately marks `TileType.Window` as walkable. Object-level collision uses `PlacedObject.blocksMovement`, not the tile type. In a zombie survival game, windows being pathable is the intended behavior — they are openings to defend.
